plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/environment/impl/NacosCustomEnvPluginService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 1999-2021 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.environment.impl;
+
+import com.alibaba.nacos.common.utils.MD5Utils;
+import com.alibaba.nacos.config.server.constant.Constants;
+import com.alibaba.nacos.plugin.environment.spi.CustomEnvironmentPluginService;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Nacos default auth plugin service implementation.
+ *
+ * @author huangtianhui
+ */
+
+public class NacosCustomEnvPluginService implements CustomEnvironmentPluginService {
```

Comment:
Should we change the plugin name like databasePasswordDecryptPluginService ?

Reply:
I don't think so，Because it's not just about encrypting the database password, it's about customizing all the environment variables